payments/tests/test_payments_mutations_accept_berth_switch_offer.py

```diff
@@ -0,0 +1,185 @@
+from payments.tests.factories import BerthSwitchOfferFactory
+from resources.schema.types import BerthNode
+from utils.relay import to_global_id
+
+ACCEPT_BERTH_SWITCH_OFFER_MUTATION = """
+mutation ACCEPT_BERTH_SWITCH_OFFER_MUTATION($input: AcceptBerthSwitchOfferMutationInput!) {
+    acceptBerthSwitchOffer(input: $input) {
+        offer {
+            application{
+                id
+                status
+            }
+            lease{
+                id
+                status
+            }
+            berth{
+                id
+            }
+            customer{
+                id
+            }
+            status
+            dueDate
+            customerFirstName
+            customerLastName
+            customerEmail
+            customerPhone
+        }
+        newLease {
+            id
+        }
+    }
+}
+"""
+
+
+@pytest.mark.parametrize(
+    "api_client", ["berth_services"], indirect=True,
+)
+@freeze_time("2020-02-01T08:00:00Z")
+def test_accept_offer(api_client):
+    customer = CustomerProfileFactory()
+    due_date = date.today() + relativedelta(days=14)
+    berth_switch_offer = BerthSwitchOfferFactory(
+        customer=customer,
+        due_date=due_date,
+        status=OfferStatus.OFFERED,
+        lease=BerthLeaseFactory(
+            customer=customer,
+            start_date=calculate_berth_lease_start_date(),
+            end_date=calculate_berth_lease_end_date(),
+            status=LeaseStatus.PAID,
+        ),
+    )
+
+    variables = {
+        "offerId": to_global_id(BerthSwitchOfferNode, berth_switch_offer.id),
+        "isAccepted": True,
+    }
+
+    executed = api_client.execute(ACCEPT_BERTH_SWITCH_OFFER_MUTATION, input=variables)
+
+    assert executed["data"]["acceptBerthSwitchOffer"]["offer"] == {
+        "application": {
+            "id": to_global_id(BerthApplicationNode, berth_switch_offer.application_id),
+            "status": berth_switch_offer.application.status.name,
+        },
+        "lease": {
+            "id": to_global_id(BerthLeaseNode, berth_switch_offer.lease_id),
+            "status": LeaseStatus.TERMINATED.name,
+        },
+        "berth": {"id": to_global_id(BerthNode, berth_switch_offer.berth_id)},
+        "customer": {"id": to_global_id(ProfileNode, berth_switch_offer.customer_id)},
+        "status": OfferStatus.ACCEPTED.name,
+        "dueDate": str(due_date),
+        "customerFirstName": berth_switch_offer.customer_first_name,
+        "customerLastName": berth_switch_offer.customer_last_name,
+        "customerEmail": berth_switch_offer.customer_email,
+        "customerPhone": berth_switch_offer.customer_phone,
+    }
+    assert executed["data"]["acceptBerthSwitchOffer"]["newLease"] == {
+        "id": to_global_id(
+            BerthLeaseNode,
+            BerthLease.objects.exclude(id=berth_switch_offer.lease_id).first().id,
+        ),
+    }
+
+
+@pytest.mark.parametrize(
+    "api_client", ["berth_services"], indirect=True,
```

Comment:
Customers would not have berth_services permissions in a real case? Are the permission issues in scope of this ticket?

Reply:
The permissions are not super clear to me what is used for what. Are they documented somewhere?

Reply:
Since it's a mutation that will be executed by customers which have not necesarily logged in, it should be a public query, so the base api client should work.